### DISCUSSION TITLE
How to change the permalink option in share chart from superset domain to embedded domain?

### DISCUSSION THREAD

**Discussion by Gokul2023-work** [apache/superset#27447] [Q&A / Help]
After embedding a dashboard in superset in to react app, if I click on share option and copy permalink, this link is pointing to the domain where I have hosted superset. I want to change this to my embedded domain within react app. My superset dashboard is used by end users for generating reports and I want this functionality entirely within my react app. 

#### How to reproduce the bug

1. Go to charts -> '...' -> share 
2. Click on copy permalink to clipboard OR share chart by email
3. The link takes me to the domain where superset is hosted.

### Expected results

I want the link to point to the embedded domain within react app

### Actual results

The link takes me to the domain where superset is hosted.

#### Screenshots

<img width="414" alt="Screenshot 2023-09-14 at 3 14 36 PM" src="https://github.com/apache/superset/assets/129675794/0a5f95f7-36e4-4796-9b38-d3a8f92167ce">

### Environment

(please complete the following information):

- browser type and version: Google chrome Version 117.0.5938.62 

- superset version: `0.0.0 - dev`
- python version: `Python 3.11.3`
- node.js version: `v20.3.0`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ✓] I have reproduced the issue with at least the latest released version of superset.
- [✓ ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by rusackas** [comment]
@yousoph @Vitor-Avila are either of you aware of such a configuration option? I haven't heard of one. Meanwhile, converting this from an Issue to a Discussion since it's not a bug, per se.

**Reply by Vitor-Avila** [reply]
I don't think this is natively possible. It would require providing a way for users to save a "mapping" of URLs similar to this:
| Dashboard link in Superstet | Dashboard link in the host app |
|--------|--------|
| www.superset.com/superset/dashboard/1 | www.embedded.com/user-dash |
| www.superset.com/superset/dashboard/2 | www.embedded.com/revenue-dash |

It would also require some changes to the client/backend so that when a guest user clicks on any sharing capability, the external link should be returned instead. Lastly, it would require changes to the SDK to be able to load the dashboard highlighting the chart they generated the link from. 

Note that the controls visible in embedded are manageable via permissions, so you might want to remove the permission to share from the role associated with your guest users. I don't think sharing has an individual permission -- it might be associated with another one so this might require some test and error.

**Comment by eram** [comment]
I'd like to revive this discussion. 
Either give me a way to disable the permalinks on embedded views or let's build it right.
My suggestion is to allow the embedding app to initialize a mapping of pages to embedded views thru the switchboard and then, when we get the embedded link from the permalink API we can replace the domain/page/tab of the permalink based on the mapping. 

![image](https://github.com/apache/superset/assets/1045377/d3ed985e-7ce7-4879-8ade-9bce1f6d8a58)
